cli/src/template_builder.rs

```diff
@@ -1728,6 +1761,54 @@ mod tests {
           |
           = Expected expression of type "Boolean", but actual type is "Integer"
         "###);
+        insta::assert_snapshot!(env.parse_err(r#"true == 1"#), @r#"
+         --> 1:9
+          |
+        1 | true == 1
+          |         ^
+          |
+          = Expected expression of type "Boolean", but actual type is "Integer"
+        "#);
+        insta::assert_snapshot!(env.parse_err(r#"true == 'a'"#), @r#"
+         --> 1:9
+          |
+        1 | true == 'a'
+          |         ^-^
+          |
+          = Expected expression of type "Boolean", but actual type is "String"
+        "#);
+        insta::assert_snapshot!(env.parse_err(r#"1 == true"#), @r#"
+         --> 1:6
+          |
+        1 | 1 == true
+          |      ^--^
+          |
+          = Expected expression of type "Integer", but actual type is "Boolean"
+        "#);
+        insta::assert_snapshot!(env.parse_err(r#"1 == 'a'"#), @r#"
+         --> 1:6
+          |
+        1 | 1 == 'a'
+          |      ^-^
+          |
+          = Expected expression of type "Integer", but actual type is "String"
+        "#);
+        insta::assert_snapshot!(env.parse_err(r#"'a' == true"#), @r#"
+         --> 1:8
+          |
+        1 | 'a' == true
+          |        ^--^
+          |
+          = Expected expression of type "String", but actual type is "Boolean"
+        "#);
+        insta::assert_snapshot!(env.parse_err(r#"'a' == 1"#), @r#"
+         --> 1:8
+          |
+        1 | 'a' == 1
+          |        ^
+          |
+          = Expected expression of type "String", but actual type is "Integer"
+        "#);
```

Comment:
nit: can you add a test for `String == Template`? It should fail, but the distinction between string and template is sometimes unclear.